Deleted relevant code of alternative object functions in Ribasim python and core (#1199)
Deltares/Ribasim: docs/core/allocation.qmd

## The optimization objective

> The goal of allocation is to get the flow to nodes with demands as close as possible to these demands. To achieve this, a sum error of terms is minimized. The form of these error terms is determined by the linear absolute object function, which is discussed further below.

Reviewer: ```suggestion
The goal of allocation is to get the flow to nodes with demands as close as possible to these demands. To achieve this, a sum error of terms is minimized.
- The form of these error terms is determined by the linear absolute object function, which is discussed further below.
```

$$
    \min E_{\text{user_demand}} + E_{\text{basin}}
$$

### User demands

> - `linear_absolute`:

Reviewer: ```suggestion
- `linear_absolute`:
```

Author: This is the same as I did

$$
    E_{\text{user_demand}} = \sum_{(i,j)\in E_S\;:\; i\in U_S} \left| F_{ij} - d_j^p(t)\right|
$$

:::{.callout-note}
When performing main network allocation, the connections to subnetworks are also interpreted as UserDemand with demands determined by subnetwork demand collection.
:::

> Linear absolute objectives cares about the actual amount of water allocated to a demand. It treats all deviations equally which means it doesn't give larger punishment if deviations increase.

Reviewer: ```suggestion
This type objective cares about the absolute amount of water allocated to a demand. It treats all deviations equally which means it doesn't give larger punishment if deviations increase.
```

I think we can get rid of the 'linear absolute' description of the chosen objective function type altogether. I chose this name to make the difference between the various options clear, but now I don't think it has any explanatory value anymore.

> :::{.callout-note}
> These options for objectives for allocation to demands have not been tested thoroughly, and might change in the future.

Reviewer: ```suggestion
-:::{.callout-note}
- These options for objectives for allocation to demands have not been tested - thoroughly, and might change in the future.
```

Reviewer: I think the callout-note below this one can also be removed

## The optimization objective

The goal of allocation is to get the flow to nodes with demands as close as possible to these demands. To achieve this, a sum error of terms is minimized.

$$
    \min E_{\text{user demand}} + E_{\text{basin}}
$$

The error between the flows and user demands is denoted by $E_{\text{user demand}}$, where
$$
    E_{\text{user demand}} = \sum_{(i,j)\in E_S\;:\; i\in U_S} \left| F_{ij} - d_j^p(t)\right|
$$

:::{.callout-note}
When performing main network allocation, the connections to subnetworks are also interpreted as UserDemand with demands determined by subnetwork demand collection.
:::

> This type of objective cares about the absolute amount of water allocated to a demand. It treats all deviations equally which means it doesn't give larger punishment if deviations increase.

Reviewer: ```suggestion
This type of objective cares about the absolute amount of water allocated to a demand. It treats all deviations equally which means it doesn't give larger punishment per flow unit if deviations increase.
```

The absolute value applied here is not supported in a linear programming context directly; this requires introduction of new variables and constraints. For more details see [here](https://optimization.cbe.cornell.edu/index.php?title=Optimization_with_absolute_values).

> :::{.callout-note}

Reviewer: This callout note can also be removed I think

In the future new optimization objectives will be introduced, for demands of basins and priorities over sources. These will be used in combination with the above, in the form of goal programming.
:::

> Likewise, the error of basin demands is the absolute value between flows consumed by basins and basin demands.

Reviewer: ```suggestion
Likewise, the error of basin demands is the absolute difference between flows consumed by basins and basin demands.
```

### Basin demands

> - `linear_absolute`:

Reviewer: ```suggestion
- `linear_absolute`:
```

$$
    E_{\text{basin}} = \sum_{i \in B_S} \left| F_i^\text{basin in} - d_i^p(t)\right|

> $$

Reviewer: I feel like the structure of this part of the documentation has to be a bit different now. Maybe the subsections `User Demands` and `Basin demands` can be removed, and you can add 2 short sentences to introduce each of the error terms (which now are for users and basins and in my PR I will add flow demands).

Author: I substituted the subsections with two sentences.
